### ISSUE TITLE
Unable to update LoadBalancer's TargetGroup: DuplicateTargetGroupName

### ISSUE THREAD

**Issue by wcurry** [kubernetes/kubernetes#96764]
**What happened**:

kube-controller-manager was unable to update an AWS LoadBalancer's TargetGroup.

**What you expected to happen**:

TargetGroup is updated without error when hosts change.

**How to reproduce it (as minimally and precisely as possible)**:

This has only happened twice so far with only one of many LoadBalancer services.

**Anything else we need to know?**:

We encountered this error while troubleshooting an issue with this service's connectivity:

```
I1120 02:58:23.948688       1 event.go:281] Event(v1.ObjectReference{Kind:"Service", Namespace:"tram-ingress", Name:"traefik-public", UID:"a9044ddf-4ac6-4a7f-9292-5e78943ca390", APIVersion:"v1", ResourceVersion:"8005791693", FieldPath:""}): type: 'Warning' reason: 'SyncLoadBalancerFailed' Error syncing load balancer: failed to ensure load balancer: error creating load balancer target group: "DuplicateTargetGroupName: A target group with the same name 'k8s-tramingr-traefikp-ca55f6909f' exists, but with different settings\n\tstatus code: 400, request id: 1b252f46-8897-421a-907e-127756658c08"
```

The error apparently has been occurring repeatedly several times a minute for over a week for this one service.

I was able to resolve the error by manually removing the listener from the load balancer in the AWS console, then deleting the target group. The target group I removed was of the name listed in the error above, but contained no targets. The new target group, created and added as a listener automatically, contained the full correct list of instances.

Later in the evening after resolving the error above we made a NodePort change to the service. The same error message reappeared and we had to manually workaround it a second time.

As of yesterday we have been rolling all ~150 nodes in our cluster. We expect the node roll and host updates to continue for several more hours today. We currently have 1588 service type LoadBalancer in this cluster. No other services have had this problem.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.17.13
- Cloud provider or hardware configuration: AWS
- OS (e.g: `cat /etc/os-release`): FCOS 32.20201018.3.0
- Kernel (e.g. `uname -a`): 5.8.15-201.fc32.x86_64

**Comment by pacoxu** [comment]
> Later in the evening after resolving the error above we made a NodePort change to the service. The same error message reappeared and we had to manually workaround it a second time.

It would be great to have a reproduce method. 

Same error message as https://github.com/kubernetes/kubernetes/issues/69264, and it may be related with the lb target group name generating logic ?

**Comment by primeroz** [comment]
> Later in the evening after resolving the error above we made a NodePort change to the service. The same error message reappeared and we had to manually workaround it a second time.

I experienced the same issue , if the node port change the update of the target groups will fail, i guess the port of a node group on aws can't be changed ?

**Comment by dgoscn** [comment]
Someone resolved this issue?

**Comment by primeroz** [comment]
Not a real `resolution` and depends on what is causing the issue for you. In my case it was an `Istio` Loadbalancer that was changing the port for some reason. I ended up _hardcoding_ the node port so to keep it to a static value

**Comment by kishorj** [comment]
@dgoscn what is your k8s version?

**Comment by dgoscn** [comment]
Hi @kishorj 

Kubernetes version is 1.17

**Comment by dgoscn** [comment]
@primeroz in my case, I tried to delete the loadbalancer server via kubectl.

`kubectl delete svc/myservice-loadbalancer`

But, it was not possible, I tried to "undo", but, it is not possible. Because of this, I still using this loadbalancer, but I can't update to a non deleting event on EKS.

**Comment by kishorj** [comment]
The issue has been fixed in v1.19 and later releases, PR https://github.com/kubernetes/kubernetes/pull/89562. 
Unfortunately, the cherry pick did not make to previous releases. Either update your k8s to 1.19 or later, or use a static nodeport assignment if on older releases.
